File: Leetcode/cpp/2210.cpp
class Solution {
public:
    int countHillValley(vector<int>& nums) {
        vector<int> output;
        for (int i = 0; i < nums.size(); i++) {
            if (output.size() == 0) {
                output.push_back(nums[i]);
            } else if (output[output.size() - 1] != nums[i]) {
                output.push_back(nums[i]);
            }
        }
        int counter = 0; 
        for (int i = 1; i < output.size() - 1; i++) {
            if ((output[i - 1] < output[i] && output[i + 1] < output[i]) || (output[i - 1] > output[i] && output[i + 1] > output[i])) {
                counter++;
            }
        }
        return counter;
    }
};

File: Leetcode/Python/_2210.py
class Solution:
    def countHillValley(self, nums: List[int]) -> int:
        unique = []
        for n in nums:
            if unique and unique[-1] != n:
                unique.append(n)
            elif not unique:
                unique.append(n)
        counter = 0
        for i in range(1, len(unique) - 1):
            if (unique[i - 1] < unique[i] and unique[i] > unique[i + 1]) or (unique[i - 1] > unique[i] and unique[i] < unique[i + 1]):
                counter += 1
        return counter
